### PR TITLE
feat(24.04): add apt and dependencies

### PR DESCRIPTION
**Pull request by lengau**

### CODE REVIEW

File: slices/dpkg.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      /etc/dpkg/dpkg.cfg:
+
+  bins:
+    essential:
+      - libbz2-1.0_libs
+      - libc6_libs
+      - liblzma5_libs
+      - libmd0_libs
+      - libselinux1_libs
+      - libzstd1
```
Comment (cjdcordeiro) [inline slices/dpkg.yaml:15]:
```suggestion
      - libzstd1_libs
```

File: slices/libsystemd0.yaml [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package: libsystemd0
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+      - libcap2_libs
+      - libgcrypt20_libs
+      - liblz4_libs
```
Comment (cjdcordeiro) [inline slices/libsystemd0.yaml:9]:
```suggestion
      - liblz4-1_libs
```

File: slices/libapt-pkg6.0.yaml [hunk 3]
```diff
@@ -0,0 +1,23 @@
+package: libapt-pkg6.0
+
+slices:
+  libs:
+    essential:
+      - libbz2-1.0_libs
+      - libc6_libs
+      - libgcc-s1_libs
+      - libgcrypt20_libs
+      - liblz4_libs
```
Comment (cjdcordeiro) [inline slices/libapt-pkg6.0.yaml:10]:
```suggestion
      - liblz4-1_libs
```

File: slices/apt.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+      /etc/apt/apt.conf.d/01-vendor-ubuntu:
+      /etc/apt/apt.conf.d/01autoremove:
+
+  bins:
+    essential:  # libseccomp2 (>= 2.4.2), libstdc++6 (>= 13.1), libsystemd0
+      - base-passwd_data
+      - libc6_libs
+      - dpkg_config
+      - gpgv_bins
+      - libapt-pkg.so.6.0_libs
```
Comment (cjdcordeiro) [inline slices/apt.yaml:15]:
```suggestion
      - libapt-pkg6.0_libs
```

File: slices/libcap2.yaml [hunk 5]
```diff
@@ -0,0 +1,9 @@
+package: libcap2
+
+slices:
+  libs:
+    essential:
+      - libc6_libs
+    contents:
+      /usr/lib/*-linux-*/libcap.so.2*:
+      /usr/lib/*-linux-*/libpsx.so.2*:
```
Comment (cjdcordeiro) [inline slices/libcap2.yaml:9]:
```suggestion
      /lib/*-linux-*/libcap.so.2*:
      /lib/*-linux-*/libpsx.so.2*:
```

File: slices/apt.yaml [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+      - ubuntu-keyring_archive
+      - libgcc-s1_libs
+      - libgnutls30_libs
+      - libseccomp2_libs
+      - libstdc++6_libs
+      - libsystemd0_libs
+    contents:
+      /usr/bin/apt*:
+      /usr/lib/apt/**:
+      /usr/lib/dpkg/methods/apt/*:
```
Comment (cjdcordeiro) [inline slices/apt.yaml:25]: this will also install `/usr/lib/dpkg/methods/apt/desc.apt`. Do we need that?

File: slices/apt.yaml [hunk 7]
```diff
@@ -0,0 +1,38 @@
...
+      /usr/lib/*-linux-*/libapt-private.so*:
```
Comment (cjdcordeiro) [inline slices/apt.yaml:26]: not a binary

**Review by cjdcordeiro — CHANGES_REQUESTED** [review]
Tyvm! Some interesting stuff here.

We'll probably need to have a deeper look at the slices design, since some of these packages (like `apt`) offer multiple functionalities

**Comment by rebornplusplus** [comment]
Hello @lengau, hope you are doing good. Any updates on this PR?

**Comment by rebornplusplus** [comment]
Hello @lengau, it has been some time since the last activity here. I see that there are some changes requested from Cris. Will you please let me know if you have some time to revisit this? Feel free to close this as well if you do not have any time. Thank you!

**Comment by lengau** [comment]
Closing this as it's tragically out of date. I'll open a new one shortly.

**Pull request closed without merging**